lib/csetgrp.gi

```diff
@@ -788,7 +788,34 @@ local a,r;
   TryNextMethod();
 end);
 
-
+InstallMethod(Intersection2, "general groups", IsIdenticalObj,
+              [IsRightCoset,IsRightCoset],
+function(cos1,cos2)
+    local H1, H2, x1, x2, shift, sigma, TheInt, CosTest, eCos, rho;
+    H1:=ActingDomain(cos1);
+    H2:=ActingDomain(cos2);
+    x1:=Representative(cos1);
+    x2:=Representative(cos2);
+    if H1=H2 then
+        if cos1=cos2 then
+            return cos1;
+        else
+            return [];
+        fi;
+    fi;
+    shift:=x1;
+    sigma:=x2 * Inverse(x1);
+    TheInt:=Intersection(H1, H2);
+    CosTest:=RightCoset(H2, sigma);
+    for eCos in RightCosets(H1, TheInt)
```

Comment:
Why `eCos`? (Not asking it to be changed, just wondering)

Reply:
I have a tendency to use names like **eCos**, **fCos** or **cos1**.
That is my style. I can change if that is a problem.